contracts/src/LightClient.sol

```diff
@@ -341,4 +381,78 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
             revert NoChangeRequired();
         }
     }
+
+    /// @notice check if more than threshold blocks passed since the last state update before
+    /// L1 Block Number
+    /// @param blockNumber The L1 block number
+    /// @param threshold The number of blocks updates to this contract is allowed to lag behind
+    function lagOverEscapeHatchThreshold(uint256 blockNumber, uint256 threshold)
+        public
+        view
+        returns (bool)
+    {
+        uint256 updatesCount = stateUpdateBlockNumbers.length;
+
+        // Handling Edge Cases
+        // Edgecase 1: The block is in the future or in the past before HotShot was live
+        if (blockNumber > block.number || updatesCount < 3) {
+            revert InsufficientSnapshotHistory();
+        }
+
+        uint256 prevBlock;
+        bool prevUpdateFound;
+
+        uint256 i = updatesCount - 1;
+        while (!prevUpdateFound) {
+            if (stateUpdateBlockNumbers[i] <= blockNumber) {
+                prevUpdateFound = true;
+                prevBlock = stateUpdateBlockNumbers[i];
+            }
+
+            // We don't consider the lag time for the first two updates
+            if (i < 2) {
```

Comment:
Why don't we consider the lag time for the first two updates?

Reply:
at one point we discussed and said that maybe there may be a lag between when the light client client contract was first deployed & instantiated, and the 1st update from a prover. So then in that instance we only consider lag times between update 2 onwards. what do you think? paging @sveitser @ImJeremyHe to hear their thoughts too

Reply:
I think this makes sense. I think the idea is that we don't expect rollups to be deployed before the prover has sent its first update to the contract. I think maybe the comment is misleading but the implementation is correct?

If a rollup has threshold 10 and we have the following history (first column is the Ethereum block number)

0 genesis
10 first LC update
12 rollup deployed
25 second LC update

Say we are now at some block after 25 and check if the lag is over the threshold at block 21.

```
blockNumber <- 21
updatesCount <- 3
i <- 2
stateUpdateBlockNumbers[2] is 25 
25 <= 21 // false
2 < 2 // false
i <- 1
stateUpdateBlockNumbers[1] is 10
10 <= 21 // true
```
So we do find the previous update.

Reply:
I see, makes sense to me!